woodwork/statistics_utils/frequency_inference/_infer_frequency.py

```diff
@@ -0,0 +1,156 @@
+import dataclasses
+
+import pandas as pd
+
+from ._clean_timeseries import _clean_timeseries
+from ._constants import FREQ_INFERENCE_THRESHOLD, WINDOW_LENGTH
+from ._determine_duplicate_values import _determine_duplicate_values
+from ._determine_extra_values import _determine_extra_values
+from ._determine_missing_values import _determine_missing_values
+from ._determine_most_likely_freq import _determine_most_likely_freq
+from ._determine_nan_values import _determine_nan_values
+from ._generate_estimated_timeseries import _generate_estimated_timeseries
+from ._generate_freq_candidates import _generate_freq_candidates
+from ._types import DataCheckMessageCode, InferDebug
+
+
+def inference_response(inferred_freq, debug_obj, debug):
+    if debug:
+        return (inferred_freq, dataclasses.asdict(debug_obj))
+    else:
+        return inferred_freq
+
+
+def infer_frequency(
```

Comment:
If this and `inference_response` are intended as public functions, they should get entries in our `api_reference.rst`